Extension/src/LanguageServer/client.ts

```diff
@@ -3506,12 +3510,40 @@ export class DefaultClient implements Client {
                     character: range.end.character,
                     line: range.end.line
                 }
-            }
+            },
+            copyToClipboard: copy ?? false
         };
 
         const result: CreateDeclarationOrDefinitionResult = await this.languageClient.sendRequest(CreateDeclarationOrDefinitionRequest, params);
-        // TODO: return specific errors info in result.
-        if (result.changes === undefined) {
+        // Create/Copy returned no result.
+        if (result.edit === undefined) {
+            await vscode.window.showInformationMessage(result.errorText); // Copy/Create Declaration/Definition was completely unsuccessful due to api failure.
+            return;
+        }
+
+        // Handle CDD error messaging
+        if (result.errorText) {
+            try {
+                await vscode.env.clipboard.writeText(result.clipboardText);
+                await vscode.window.showInformationMessage(result.errorText + " Declaration/definition was copied to clipboard."); // CDD failed but fallback to clipboard was succesful
+
+            } catch {
+                // handle failure
+                await vscode.window.showInformationMessage(result.errorText); // CDD failed but fallback was unsuccessful.
+            }
+            return;
+        }
+
+        // Handle copy to clipboard.
+        if (params.copyToClipboard) {
+            if (result.clipboardText) {
+                try {
+                    await vscode.env.clipboard.writeText(result.clipboardText);
+                } catch {
+                    // handle failure
```

Comment:
I don't see anyone else on GitHub that has a try catch specifically around `await vscode.env.clipboard.writeText`, so I'd prefer if this were removed, since it's unnecessary.

Reply:
I would prefer to keep it